libcloud/compute/drivers/kubevirt.py

```diff
@@ -0,0 +1,996 @@
+            secure = True
+
+        if host.startswith('https://'):
+            secure = True
+
+        # strip the prefix
+        prefixes = ['http://', 'https://']
+        for prefix in prefixes:
+            if host.startswith(prefix):
+                host = host.lstrip(prefix)
+
+        super(KubeVirtNodeDriver, self).__init__(key=key,
+                                                 secret=secret,
+                                                 secure=secure,
+                                                 host=host,
+                                                 port=port,
+                                                 key_file=key_file,
+                                                 cert_file=cert_file)
+
+        # check if both key and cert files are present
+        if key_file or cert_file:
+            if not(key_file and cert_file):
+                raise Exception("Both key and certificate files are needed")
+
+        if ca_cert:
+            self.connection.connection.ca_cert = ca_cert
+        else:
+            # do not verify SSL certificate
+            self.connection.connection.ca_cert = False
+
+        self.connection.secure = secure
+        self.connection.host = host
+        self.connection.port = port
+
+        if self.connectionCls == KubernetesConnection:
+            self.connection.secret = secret
+        self.connection.key = key
+
+    def list_nodes(self, namespace=None):
+        namespaces = []
+        if namespace:
+            namespaces.append(namespace)
+        else:
+            for ns in self.list_locations():
+                namespaces.append(ns.name)
+
+        dormant = []
+        live = []
+        for ns in namespaces:
+            req = KUBEVIRT_URL + 'namespaces/' + ns + \
+                "/virtualmachines"
+            result = self.connection.request(req)
+            if result.status != 200:
+                continue
+            result = result.object
+            for item in result['items']:
+                if not item['spec']['running']:
+                    dormant.append(item)
+                else:
+                    live.append(item)
+        vms = []
+        for vm in dormant:
+            vms.append(self._to_node(vm, is_stopped=True))
+
+        for vm in live:
+            vms.append(self._to_node(vm, is_stopped=False))
+
+        return vms
+
+    def get_node(self, id=None, name=None):
+        "get a vm by name or id"
+        if not id and not name:
+            raise ValueError("This method needs id or name to be specified")
+        nodes = self.list_nodes()
+        if id:
+            node_gen = filter(lambda x: x.id == id,
+                              nodes)
+        if name:
+            node_gen = filter(lambda x: x.name == name,
+                              nodes)
+
+        try:
+            return next(node_gen)
+        except StopIteration:
+            raise ValueError("Node does not exist")
+
+    def ex_start_node(self, node):
```

Comment:
This method name can now be changed to ``start_node()`` (see https://github.com/apache/libcloud/pull/1375 for details).